From 295d6bd776bb084464ced877f7277d5c61f9cb83 Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lauri@hacktheplanet.fi>
Date: Fri, 11 Apr 2025 00:24:18 +0900
Subject: [PATCH] sysfs: mount /sys/fs/bpf if available

eg. cilium expects /sys/fs/bpf to be mounted. systemd mounts it here as
well.
---
 init.d/sysfs.in | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/init.d/sysfs.in b/init.d/sysfs.in
index 7ef06142e..402c3435e 100644
--- a/init.d/sysfs.in
+++ b/init.d/sysfs.in
@@ -105,6 +105,15 @@ mount_misc()
 			efivarfs /sys/firmware/efi/efivars 2> /dev/null
 		eend 0
 	fi
+
+	# set up kernel support for bpf filesystem
+	if [ -d /sys/fs/bpf ] && ! mountinfo -q /sys/fs/bpf; then
+		if grep -qs 'bpf$' /proc/filesystems; then
+			ebegin "Mounting bpf filesystem"
+			mount -n -t bpf -o ${sysfs_opts} bpf /sys/fs/bpf
+			eend $?
+		fi
+	fi
 }
 
 restorecon_sys()
